kroxylicious/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -112,65 +135,121 @@ else if (requestFilterResult.message() != null) {
                 LOGGER.warn("Unexpected message writing to upstream: {}", msg, new IllegalStateException());
             }
             ctx.write(msg, promise);
+            return CompletableFuture.completedFuture(null);
+        }
+    }
+
+    private <T> CompletableFuture<T> switchBackToNettyIfRequired(ChannelHandlerContext ctx, CompletableFuture<T> future) {
+        CompletableFuture<T> nettyDrivenFuture = new CompletableFuture<>();
```

Comment:
I'm not grasping why completing the future has to be done on the netty thread.

Reply:
Might drop this commit as being surplus to requirement and come back to it. 

I was speculating it could be kinder to netty to switch back once instead of making multiple calls to the netty context as we drain the queue, each call getting put into the executor queue. But that needs testing.

Maybe a better argument is it could make increase our confident that the work of the FilterHandler is executed single-threaded (though the queue should enforce that), and puts the framework in control of where kroxylicious-owned logic is executed.

Reply:
I think splitting this change out is the right call. As I'm not entirely sure `ctx.executor().execute()` is going to take us back to the originating thread or if its just going to cause more context thrashing. It probably warrants a benchmark (we can craft one)

Reply:
Reverted the commit, so it's removed from this PR :+1: